hs_tools_resource/models.py

```diff
@@ -85,16 +85,28 @@ def get_supported_res_types_str(self):
         return ','.join([parameter.description for parameter in self.supported_res_types.all()])
 
     @classmethod
+    def _add_supported_res_type(cls, meta_instance, res_type):
+        if isinstance(res_type, int):
+            # "copy res" or "create a new version"
+            qs = SupportedResTypeChoices.objects.filter(id=res_type)
+            if not qs.exists():
+                raise
+            meta_instance.supported_res_types.add(qs[0])
+        else:
+            # create or update res
+            qs = SupportedResTypeChoices.objects.filter(description__iexact=res_type)
+            if qs.exists():
+                meta_instance.supported_res_types.add(qs[0])
+            else:
+                meta_instance.supported_res_types.create(description=res_type)
```

Comment:
is there a validation function created to validate the res_type values? For example, take a look on this validation function for SWAT objectives: https://github.com/hydroshare/hydroshare/blob/develop/hs_swat_modelinstance/models.py#L99